### PR TITLE
[NFC] Move DIExpressionCursor to DebugInfoMetadata.h

### PR DESCRIPTION
**Pull request by rastogishubham**
This is an NFC patch to move DIExpressionCursor to DebugInfoMetada.h, so that it can be used by classes in that header file.

Specifically, I want to use DIExpressionCursor in a subsequent patch:
https://github.com/llvm/llvm-project/pull/71718

### CODE REVIEW

**Review by dwblaikie — APPROVED** [review]
Simple enough (maybe wait on committing this until the usage is approved - or roll it back if the usage doesn't eventuate (if subsequent patches go in a different direction, etc))

**Review by adrian-prantl — APPROVED** [review]

**Comment by MaskRay** [comment]
@rastogishubham This has been approved long ago. Are you planning to re-run the tests (since old results are stale) and land it?

**Review by MaskRay — APPROVED** [review]

**Comment by rastogishubham** [comment]
@MaskRay I have subsequent patches that build on top of this change that I had to stop working on because of some other work so I didn't merge this patch as well, do you suggest I merge this patch or wait?

**Pull request merged**
